src/lib/worker.js

```diff
@@ -1,42 +1,83 @@
 import path from 'path';
+import {spawn} from 'child_process';
 
-const testOptions = JSON.parse(process.env.testOptions);
-const cucumber = require(testOptions.cucumberPath).Cli;
-
-const featureFile = process.env.featureFile;
-const featureFileData = path.parse(process.env.featureFile);
-const featureName = featureFileData.name;
-const scenarioLine = process.env.scenarioLine;
-const logFileName = featureName + '-line-' + scenarioLine + '.json';
-const logFile = path.join(testOptions.logDir, logFileName);
-const relativeLogFile = path.relative(process.cwd(), logFile);
-
-let args = ['', '', featureFile + ':' + scenarioLine, '-f', 'json:' + relativeLogFile];
-
-testOptions.requires.forEach(function(arg) {
-  args.push('-r');
-  args.push(arg);
-});
-
-let startTime = new Date();
-
-try {
-  cucumber(args).run(function(isSuccessful) {
-    let exitCode = (isSuccessful) ? 0 : 1;
-    process.send({
-      type: 'result',
-      exitCode: exitCode,
-      resultFile: logFile,
-      duration: new Date() - startTime
+export default class Worker {
+  constructor(options) {
+    this.options = options;
+  }
+
+  execute(callback) {
+    let featureFile = this.options.featureFile;
+    let featureFileData = path.parse(featureFile);
+    let featureName = featureFileData.name;
+    let scenarioLine = this.options.scenarioLine;
+    let logFileName = featureName + '-line-' + scenarioLine + '.json';
+    let logFile = path.join(this.options.logDir, logFileName);
+    let relativeLogFile = relativeLogFile = path.relative(process.cwd(), logFile);
+    let cucumberPath = this.options.cucumberPath;
+    let args = [featureFile + ':' + scenarioLine, '-f', 'json:' + relativeLogFile];
+    let running = false;
+
+    this.options.requires.forEach(function(arg) {
+      args.push('-r');
+      args.push(arg);
+    });
+
+    // TODO: allow inheriting stdio
+    this.child = spawn(
+      cucumberPath,
```

Comment:
recommend spawning using `process.execPath` to play nicely as a library, so you use the same exact node executable as the user of this lib.